runtime-light/utils/logs.h

```diff
@@ -24,25 +25,44 @@ namespace kphp::log {
 
 namespace impl {
 
+template<typename T>
+requires std::is_floating_point_v<std::remove_cvref_t<T>>
+struct floating_wrapper final {
+  T value{};
+};
+
+template<typename T>
+constexpr auto wrap_log_argument(T&& t) noexcept -> decltype(auto) {
+  if constexpr (std::is_floating_point_v<std::remove_cvref_t<T>>) {
+    return impl::floating_wrapper<std::remove_cvref_t<T>>{.value = std::forward<T>(t)};
+  } else {
+    return std::forward<T>(t);
+  }
+}
+
+template<typename T>
+using wrapped_arg_t = std::invoke_result_t<decltype(impl::wrap_log_argument<T>), T>;
+
 enum class level : size_t { error = 1, warn, info, debug, trace };
 
 template<typename... Args>
-void log(level level, const std::format_string<Args...>& fmt, Args&&... args) noexcept {
+void log(level level, std::format_string<impl::wrapped_arg_t<Args>...> fmt, Args&&... args) noexcept {
   if (std::to_underlying(level) > k2::log_level_enabled()) {
     return;
   }
 
   static constexpr size_t LOG_BUFFER_SIZE = 512;
   std::array<char, LOG_BUFFER_SIZE> log_buffer{};
-  const auto [out, size]{std::format_to_n(log_buffer.data(), log_buffer.size() - 1, fmt, std::forward<Args>(args)...)};
+  const auto [out, size]{std::format_to_n<decltype(log_buffer.data()), impl::wrapped_arg_t<Args>...>(log_buffer.data(), log_buffer.size() - 1, fmt,
+                                                                                                     impl::wrap_log_argument(std::forward<Args>(args))...)};
   *out = '\0';
   k2::log(std::to_underlying(level), size, log_buffer.data());
 }
 
 } // namespace impl
 
 template<typename... Args>
-[[noreturn]] void error(const std::format_string<Args...>& fmt, Args&&... args) noexcept {
+[[noreturn]] void error(std::format_string<impl::wrapped_arg_t<Args>...> fmt, Args&&... args) noexcept {
```

Comment:
`fmt` was passed by const reference, why for now it isn't?

Reply:
I decided to pass it the same way it's passed to `std::format` -- by value. It's ok since it's just a wrapper around `std::string_view`